geopandas/io/tests/test_file.py

```diff
@@ -1136,6 +1139,62 @@ def test_write_read_file(test_file, engine):
     os.remove(os.path.expanduser(test_file))
 
 
+def test_to_file__metadata(tmpdir, df_points, engine):
```

Comment:
I'd prefer this to be split into 4 test cases: Fiona with and without supported version, pyogrio with and without supported version.

Something like (untested)

```Python
@pytest.mark.skipif(not Fiona, reason="Fiona not available")
@pytest.mark.skipif(FIONA_GE_19, reason="Fiona >= 1.9 supports metadata")
def test_to_file_metadata_unsupported_fiona_version(tmpdir, df_points, engine):
    with pytest.raises(...):
        df_points.to_file(
            tempfilename, driver="GPKG", engine="fiona", metadata=metadata
         )

@pytest.mark.skipif(not Fiona, reason="Fiona not available")
@pytest.mark.skipif(not FIONA_GE_19, reason="Fiona >= 1.9 required to support metadata")
def test_to_file_metadata_supported_fiona_version(tmpdir, df_points, engine):
    ...
```

Not all of these will be tested in all of the CI environments, but this will hopefully make more explicit which ones are actually tested vs skipped.